test/unittest/tests/test_bounds_restore.cpp

```diff
@@ -0,0 +1,63 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+
+#include "utils.h"
+#include "wiredtiger.h"
+#include "wrappers/connection_wrapper.h"
+#include "wt_internal.h"
+
+int
+validate_cursor_bounds_restore(WT_CURSOR *cursor, uint64_t original_cursor_flags)
+{
+    assert(cursor->flags == original_cursor_flags);
+    return (0);
+}
+
+TEST_CASE("Bounds save and restore flag logic", "[bounds_restore]")
+{
+    uint64_t original_cursor_flags;
```

Comment:
In C++, it's best to declare variables as late as possible. So, an option is to remove this line, and then change

`original_cursor_flags = mock_cursor.flags;`

below to

`uint64_t  original_cursor_flags = mock_cursor.flags;`

Reply:
[0cb5d18](https://github.com/wiredtiger/wiredtiger/pull/8335/commits/0cb5d1872bce15272b3e941cf856b34d45d8d75c)